test/session-pool.ts

```diff
@@ -1349,6 +1365,84 @@ describe('SessionPool', () => {
       });
     });
   });
+
+  describe('trace annotations on active span', () => {
+    const exporter = new InMemorySpanExporter();
+    const globalProvider = new NodeTracerProvider({
+      sampler: new AlwaysOnSampler(),
+      exporter: exporter,
+    });
+    globalProvider.addSpanProcessor(new SimpleSpanProcessor(exporter));
+    globalProvider.register();
+
+    const contextManager = new AsyncHooksContextManager();
+    setGlobalContextManager(contextManager);
+
+    beforeEach(() => {
+      sessionPool.isOpen = true;
+      sessionPool._isValidSession = () => true;
+    });
+
+    afterEach(async () => {
+      exporter.forceFlush();
+      exporter.reset();
+      globalProvider.forceFlush();
+    });
+
+    after(async () => {
+      disableContextAndManager(contextManager);
+      await globalProvider.shutdown();
+    });
+
+    it('annotations when acquiring a session', () => {
+      const topLevelSpanName = 'testSessionPool.acquire';
+      startTrace(topLevelSpanName, {}, async span => {
+        const fakeSession = createSession();
+        const now = Date.now();
+
+        const stub = sandbox
+          .stub(sessionPool, '_getSession')
+          .resolves(fakeSession);
+        const session = await sessionPool._acquire();
+        const [startTime] = stub.getCall(0).args;
+
+        assert(isAround(startTime, now));
+        assert.strictEqual(session, fakeSession);
+
+        await sessionPool._release(session);
+        span.end();
+
+        const spans = exporter.getFinishedSpans();
+        assert.strictEqual(
+          spans.length,
+          1,
+          'Exactly 1 span should have been exported'
+        );
+        assert.strictEqual(
+          spans[0].name,
+          `${SPAN_NAMESPACE_PREFIX}.${topLevelSpanName}`,
+          'Expected only the top-level created span'
+        );
+        const span0 = spans[0];
```

Comment:
I don't see any assert on the annotations. 

Ideally we should add below tests 

1) Executing a query when no sessions are available

-> These will have "Acquiring Session", "Waiting for session to become available", "Creating Session" , "Acquired session" annotation

2) Executing a query when session is available"

 These will have "Acquiring Session", "Cache hit: has usable session" , "Acquired session" annotation

Reply:
Refer https://github.com/googleapis/java-spanner/blob/main/google-cloud-spanner/src/test/java/com/google/cloud/spanner/OpenTelemetrySpanTest.java

Reply:
There are assertions on the annotations for SessionPool,. Also in our discussions, we agreed that a follow-up PR this one or two shall be a PR to add tests against the Spanner Mock server which test the whole grouping hence we'll able to test out all the annotations.